core/src/test/java/org/apache/iceberg/rest/responses/TestErrorResponseParser.java

```diff
@@ -40,14 +41,25 @@ public void testErrorResponseToJson() {
   public void testErrorResponseFromJson() {
     String message = "The given namespace does not exist";
     String type = "NoSuchNamespaceException";
-    Integer code = 404;
+    int code = 404;
     String errorModelJson = String.format("{\"message\":\"%s\",\"type\":\"%s\",\"code\":%d}", message, type, code);
     String json = "{\"error\":" + errorModelJson + "}";
 
     ErrorResponse expected = ErrorResponse.builder().withMessage(message).withType(type).responseCode(code).build();
     assertEquals(expected, ErrorResponseParser.fromJson(json));
   }
 
+  @Test
+  public void testErrorResponseFromJsonWithoutCode() {
+    String message = "The given namespace does not exist";
+    String type = "NoSuchNamespaceException";
+    String errorModelJson = String.format("{\"message\":\"%s\",\"type\":\"%s\"}", message, type);
+    String json = "{\"error\":" + errorModelJson + "}";
+
+    AssertHelpers.assertThrows("ErrorResponse should contain code", IllegalArgumentException.class,
+        "Cannot parse missing int code", () -> ErrorResponseParser.fromJson(json));
+  }
```

Comment:
My concern here is that the error code can still be parsed from other places (like the response itself). I agree it _should_ be there, but I'm not sure if adding this additional constraint provides that much benefit.

The number of boxed `Integer`s is still likely to be relatively small, as most operations will still go through normal `FileIO`.

I think this optimization is possibly a bit premature, but I'll let others weigh in as well first.

Reply:
> I'm not sure if adding this additional constraint provides that much benefit.

+1, I also expected that code should be present anyways, absence of it during build() time we check code is not null for creating Response obj, and throw [IllegalArgs exception](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponse.java#L132).

> I think this optimization is possibly a bit premature, but I'll let others weigh in as well first.

Agree with you, I don't have a strong opinion of must having this change either. Happy to close this as well :) !!!